src/Cryptol/ModuleSystem/Names.hs

```diff
@@ -16,14 +16,22 @@ import Cryptol.ModuleSystem.Name
 data Names = One Name | Ambig (Set Name) -- ^ Non-empty
   deriving (Show,Generic,NFData)
 
+-- | The returned list of names will be non-empty.
 namesToList :: Names -> [Name]
 namesToList xs =
   case xs of
     One x -> [x]
     Ambig ns -> Set.toList ns
 
 anyOne :: Names -> Name
-anyOne = head . namesToList
+anyOne xs =
+  case xs of
+    One x -> x
+    Ambig ns
+      |  x:_ <- Set.toList ns
```

Comment:
Rather than converting the Set to a List and then only taking the first element (which is probably lazily cheap) would this be more direct as something like `Ambig ns -> if Set.null ns then panic .. else Set.elemAt 0 ns` ?

Reply:
Sure, I think your version works just as well. I'll change it.